Useridtracking

gyulong1 opened the pull request:
Updated user_id tracking without a handler file

Error handling now in queries.js

gyulong1 committed:
Updated userid tracking functions

Username is updated with each message sent.

getUsername retrieves username for a userid.

gyulong1 committed:
Updated user_id tracking

Updated user_id tracking without a handler

Error handling in queries.js

nicktohzyu reviewed (approved):
Approved, however I did not test with postgres. Usage of these helpers will require updating the database

nicktohzyu commented on db/queries.js:
```diff
@@ -198,6 +198,41 @@ module.exports.getItemName = async function (params, callback) {
+        let args;
+        //checks if username is empty
+        if (params.user_name === undefined) {
+           args = ["", params.user_id, ""];
```
3 spaces for indentation ‽

nicktohzyu replied:
I'd suggest to `params.user_name = ""` instead. This makes it act like a guard clause. Then the line inside the else block will be moved outside (and the else block will be empty, hence remove it)

gyulong1 replied:
done, thanks

nicktohzyu commented on db/queries.js:
```diff
@@ -198,6 +198,41 @@ module.exports.getItemName = async function (params, callback) {
+        //checks if username is empty
+        if (params.user_name === undefined) {
+           args = ["", params.user_id, ""];
+        } else{
```
Please fix brace spacing - use the automatic formatting tool

gyulong1 replied:
done

nicktohzyu commented on db/queries.js:
```diff
@@ -198,6 +198,41 @@ module.exports.getItemName = async function (params, callback) {
+		miscellaneous.usernames (user_name, user_id)
+		values ($1, $2)
+        on conflict (user_id) do update
+		set user_name = $3;`;
```
Would $1 work instead of $3?

gyulong1 replied:
yes it does work, thanks

nicktohzyu commented:
Please update on receiving a callback as well. In `index.js`, line 92

gyulong1 committed:
Updated useridtracking

Changed implementation for updateUsername function

Now updates username on callback

Fixed formatting issues
